src/main/java/de/tum/in/www1/artemis/repository/CourseRepository.java

```diff
@@ -37,36 +37,34 @@ public interface CourseRepository extends JpaRepository<Course, Long> {
     @Query("select distinct course.teachingAssistantGroupName from Course course")
     Set<String> findAllTeachingAssistantGroupNames();
 
-    @Query("select distinct course from Course course where course.instructorGroupName like :#{#name}")
+    @Query("select distinct course from Course course where course.instructorGroupName like :name")
     Course findCourseByInstructorGroupName(@Param("name") String name);
 
-    @Query("select distinct course from Course course where course.editorGroupName like :#{#name}")
+    @Query("select distinct course from Course course where course.editorGroupName like :name")
     Course findCourseByEditorGroupName(@Param("name") String name);
 
-    @Query("select distinct course from Course course where course.teachingAssistantGroupName like :#{#name}")
+    @Query("select distinct course from Course course where course.teachingAssistantGroupName like :name")
     Course findCourseByTeachingAssistantGroupName(@Param("name") String name);
 
-    @Query("select distinct course from Course course where course.studentGroupName like :#{#name}")
+    @Query("select distinct course from Course course where course.studentGroupName like :name")
     Course findCourseByStudentGroupName(@Param("name") String name);
 
     @Query("""
             SELECT DISTINCT c FROM Course c
-            WHERE (c.startDate <= :now
-            	OR c.startDate IS NULL)
-            AND (c.endDate >= :now
-            	OR c.endDate IS NULL)
+            WHERE (c.startDate <= :now OR c.startDate IS NULL)
+                AND (c.endDate >= :now OR c.endDate IS NULL)
             """)
     List<Course> findAllActive(@Param("now") ZonedDateTime now);
 
     // Note: you should not add exercises or exercises+categories here, because this would make the query too complex and would take significantly longer
-    @EntityGraph(type = LOAD, attributePaths = { "lectures", "lectures.attachments", "exams" })
+    @EntityGraph(type = LOAD, attributePaths = { "lectures", "lectures.attachments" })
     @Query("""
             SELECT DISTINCT c
             FROM Course c
             WHERE (c.startDate <= :now OR c.startDate IS NULL)
                 AND (c.endDate >= :now OR c.endDate IS NULL)
             """)
-    List<Course> findAllActiveWithLecturesAndExams(@Param("now") ZonedDateTime now);
+    List<Course> findAllActiveWithLectures(@Param("now") ZonedDateTime now);
```

Comment:
What is the reason behind this and similar changes? If I see it correctly, on all occurrences, you add the exams anyway so the only thing you gain is an additional DB call.

Reply:
We did not find any easy way to include the exams while still filtering them correctly and also including courses where all exams where filtered out

Reply:
Ah, I get it. Get all vs get some specific. That makes sense.